vaadin-cdi/src/main/java/com/vaadin/cdi/context/RouteScopedContext.java

```diff
@@ -151,6 +151,13 @@ private void relocate(UI ui, RouteStorageKey key) {
                     new RouteStorageKey(key.getOwner(), getUIStoreId(ui)));
         }
 
+        private List<ContextualStorage> getActiveContextualStorages() {
+            return getKeySet().stream().filter(
+                    key -> key.getUIId().equals(getUIStoreId(UI.getCurrent())))
+                    .map(key -> getContextualStorage(key, false))
```

Comment:
Shouldn't this method also check that the route's owner is in the current navigation chain?

Reply:
route's owner of WHAT ?
route's owner is something that defines a bean . There is no any bean in the context.
The method name is `getActiveContextualStorages` which mean : get all active storages. And there is no beans, injection points , etc.
The `UI` contains an active navigation chain. Based on this chain all the storages which has a key associated with the active `UI` instance should be returned.
I don't even understand your question.

Reply:
`ContextualStorage`s of route scope are associated with corresponding `RouteStorageKey` which includes the owner of the bean and the UI id this bean belongs to (or, more precisely, it's lifespan belong to). That's why I asked about the owner, owner of a bean. But I believe `getActiveContextualStorages` is called in the codes which don't care about the owner of a bean (I found only one place - in `destroyAllActive`), and, thus, you don't use it.
Anyway, thanks for explanations.